kratos/sources/initial_state.cpp

```diff
@@ -0,0 +1,180 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Alejandro Cornejo
+//
+//
+
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/initial_state.h"
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Classes
+///@{
+
+
+    /// Only defining Dimension constructor.
+    InitialState::InitialState(const SizeType Dimension)
+    {
+        const SizeType voigt_size = (Dimension == 3) ? 6 : 3;
+        mInitialStressVector.resize(voigt_size, false);
+        mInitialStrainVector.resize(voigt_size, false);
+        mInitialDeformationGradientMatrix.resize(Dimension, Dimension, false);
+
+        noalias(mInitialStressVector) = ZeroVector(voigt_size);
+        noalias(mInitialStrainVector) = ZeroVector(voigt_size);
+        noalias(mInitialDeformationGradientMatrix) = ZeroMatrix(Dimension, Dimension);
+    }
+
+    // Full constructor
+    InitialState::InitialState(const Vector& rInitialStrainVector,
+                    const Vector& rInitialStressVector,
+                    const Matrix& rInitialDeformationGradientMatrix)
+    {
+        const SizeType voigt_size = rInitialStrainVector.size();
+        const SizeType dimension  = rInitialDeformationGradientMatrix.size1();
+
+        KRATOS_ERROR_IF(voigt_size <= 0 || dimension <= 0) << "The imposed vector or matrix is null..." << std::endl;
+
+        mInitialStressVector.resize(voigt_size, false);
+        mInitialStrainVector.resize(voigt_size, false);
+        mInitialDeformationGradientMatrix.resize(dimension, dimension, false);
+
+        noalias(mInitialStressVector) = rInitialStressVector;
+        noalias(mInitialStrainVector) = rInitialStrainVector;
+        noalias(mInitialDeformationGradientMatrix) = rInitialDeformationGradientMatrix;
+    }
+
+    // Selective constructor for vectors
+    InitialState::InitialState(const Vector& rImposingEntity,
+                    const int InitialImposingType = static_cast<int>(InitialImposingType::STRAIN_ONLY))
```

Comment:
while looking at the CentOS error I noticed all the casts here
@AlejandroCornejo those should not be there, you can directly work with the `enum`
can you please fix this in a follow up PR?